[quant] QNNPACK factoring out common structs from conv/deconv

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37497 [quant] QNNPACK factoring out common structs from conv/deconv**
* #36790 [quant] QNNPACK deconv kernel and tests
* #37405 [quant] QNNPACK deconvolution packing



Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

Commit:
[quant] QNNPACK factoring out common structs from conv/deconv

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"


Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"


Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"


Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"


Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"


Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

supriyar reviewed (approved):
Looks good, thanks for refactoring

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK factoring out common structs from conv/deconv"




Differential Revision: [D21301244](https://our.internmc.facebook.com/intern/diff/D21301244)

[ghstack-poisoned]
